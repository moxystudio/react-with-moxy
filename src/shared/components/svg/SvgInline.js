import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import styles from './Svg.css';

const Svg = (props) => {
    const { svg, className, ...otherProps } = props;

    return (
        <i
            className={ classnames(styles.svg, className) }
            { ...otherProps }
            dangerouslySetInnerHTML={ { __html: svg } } />
    );
};

Svg.propTypes = {
    svg: PropTypes.string.isRequired,
    className: PropTypes.string,
};

export default Svg;
