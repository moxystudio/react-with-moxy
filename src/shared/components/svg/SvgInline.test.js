import React from 'react';
import { shallow } from 'enzyme';
import SvgInline from './SvgInline';
import moxySquareSvg from 'shared/media/images/logos/moxy-square.inline.svg';

it('should render an inline SVG correctly', () => {
    const tree = shallow(<SvgInline svg={ moxySquareSvg } />);

    expect(tree).toMatchSnapshot();
});

it('should render with the specified className', () => {
    const tree = shallow(<SvgInline svg={ moxySquareSvg } className="foo" />);

    expect(tree).toMatchSnapshot();
});
