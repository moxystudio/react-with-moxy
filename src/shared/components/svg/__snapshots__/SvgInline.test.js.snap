// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an inline SVG correctly 1`] = `
<i
  className="svg"
  dangerouslySetInnerHTML={
    Object {
      "__html": "moxy-square.inline.svg",
    }
  }
/>
`;

exports[`should render with the specified className 1`] = `
<i
  className="svg foo"
  dangerouslySetInnerHTML={
    Object {
      "__html": "moxy-square.inline.svg",
    }
  }
/>
`;
